modules/fedimint-ln-client/src/pay.rs

```diff
@@ -194,35 +210,22 @@ impl LightningPayFunded {
         let gateway = self.gateway.clone();
         let payload = self.payload.clone();
         let contract_id = self.payload.contract_id;
-        let timeout_context = global_context.clone();
-        vec![
-            StateTransition::new(
-                // Wait for the result of the payment by the gateway
-                Self::await_outgoing_contract_execution(gateway, payload),
-                move |_dbtx, result, old_state| {
-                    Box::pin(Self::transition_outgoing_contract_execution(
-                        old_state,
-                        timeout_context.clone(),
-                        result,
-                        contract_id,
-                    ))
-                },
-            ),
-            // wait for gatewayd for two minutes before timing out
-            StateTransition::new(
-                sleep(Duration::from_secs(120)),
-                move |_dbtx, (), old_state| {
-                    Box::pin(Self::transition_pay_timeout(
-                        old_state,
-                        global_context.clone(),
-                        contract_id,
-                    ))
-                },
-            ),
-        ]
+        vec![StateTransition::new(
+            // Immediately try to pay the invoice by contacting the gateway
```

Comment:
It can if we want it to. The thinking behind making it a separate state is if the client crashes/stops after awaiting the Lightning Outcome, but before trying to pay the invoice, we won't need to await the Lightning Outcome again (if it was in the same transition function we'd need to do the whole thing again).

I think it's also a bit more user friendly just to know when the offer has been accepted, but not executed yet

Reply:
As Eric mentioned below, this should ideally move back to the trigger function once we have Client NG on the gateway side.